src/libstd/collections/hash/map.rs

```diff
@@ -2445,13 +2445,18 @@ impl RandomState {
         // iteration order allows a form of DOS attack. To counter that we
         // increment one of the seeds on every RandomState creation, giving
         // every corresponding HashMap a different iteration order.
-        thread_local!(static KEYS: Cell<(u64, u64)> = {
-            Cell::new(sys::hashmap_random_keys())
+        thread_local!(static KEYS: Cell<[u64; 2]> = {
+            let mut buf = [0u8; 16];
+            // we consciously ignore potential errors here
+            let _ = getrandom::getrandom(&mut buf);
+            // SAFETY: it's safe to transmute random 16 bytes to two `u64`s
+            let buf: [u64; 2] = unsafe { core::mem::transmute(buf) };
```

Comment:
is there a way to change this into something safe too?

Reply:
IIUC not without calling `getrandom` twice

Reply:
IIUC this isn't safe — a `u8` buffer doesn't guarantee sufficient alignment. Instead I think you should do something like [this](https://github.com/rust-random/rand/blob/master/rand_isaac/src/isaac64.rs#L301).
Maybe @RalfJung knows best?

Or you could use two calls to `from_ne_bytes` as in the windows pipe code below. Safe and probably no significant performance issues.

Reply:
I thought new variables are always properly aligned, no? To be safe we can use `transmute_copy` it does not have an alignment requirement [anymore](https://github.com/rust-lang/rust/pull/55052).

Reply:
This transmutes the values, not references to the values, so there shouldn't be a problem with alignment.

Reply:
However, the comment maybe should stress that the "random 16 bytes" are *initialized*. "random" is sometimes used to refer to the content of uninitialized storage, and that would not be sufficient.